sky/spot/spot_utils.py

```diff
@@ -513,24 +513,44 @@ def is_spot_controller_up(
     stopped_message: str,
 ) -> Tuple[Optional[global_user_state.ClusterStatus],
            Optional[backends.Backend.ResourceHandle]]:
+    """Check if the spot controller is up.
+
+    It can be used to check the actual controller status (since the autostop is
+    set for the controller) before the spot commands interact with the
+    controller.
+
+    Returns:
+        controller_status: The status of the spot controller. If it fails during
+          refreshing the status, it will be the cached status. None if the
+          controller does not exist.
+        handle: The ResourceHandle of the spot controller. None if the
+          controller is not UP or does not exist.
+
+    Raises:
+        exceptions.ClusterOwnerIdentityMismatchError: if the current user is not
+          the same as the user who created the cluster.
+        exceptions.CloudUserIdentityError: if we fail to get the current user
+          identity.
+    """
     try:
         controller_status, handle = backend_utils.refresh_cluster_status_handle(
             SPOT_CONTROLLER_NAME, force_refresh=True)
-    except (exceptions.CloudUserIdentityError,
-            exceptions.ClusterOwnerIdentityMismatchError,
-            exceptions.ClusterStatusFetchingError) as e:
+    except exceptions.ClusterStatusFetchingError as e:
+        # We do not catch the exceptions related to the cluster owner identity
+        # mismatch, please refer to the comment in
+        # `backend_utils.check_cluster_available`.
         logger.warning(
             f'Failed to get the status of the spot controller. '
-            'It is not fatal, but spot commands may hang, when '
-            'the controller is not up, or have inaccurate statement.\n'
-            f'Details: {e}')
+            'It is not fatal, but spot commands/calls may hang or make '
+            'inaccurate statement, when the controller is not up.\n'
```

Comment:
RE: https://github.com/skypilot-org/skypilot/pull/1513#discussion_r1051685347

I feel this can be a bit hard to understand. How about changing to "or return stale information"?

Reply:
Good point! Changed. Thanks!